SecretSanta/test/SecretSanta.Web.UITests/UserPageTests.cs

```diff
@@ -0,0 +1,203 @@
+                IWebElement editLink = page.GetEditLink(oldFullName);
+                editLink.Click();
+                EditUsersPage editUsersPage = new EditUsersPage(Driver);
+                editUsersPage.UserFirstNameTextBox.Clear();
+                editUsersPage.UserFirstNameTextBox.SendKeys(newUserFirstName);
+                editUsersPage.UserLastNameTextBox.Clear();
+                editUsersPage.UserLastNameTextBox.SendKeys(newUserLastName);
+                editUsersPage.SubmitButton.Click();
+
+                // Ensure old username is removed and new user name exists
+                userNames = page.UserNames;
+                Assert.IsFalse(userNames.Contains(oldFullName));
+                Assert.IsTrue(userNames.Contains(newFullName));
+            });
+        }
+
+        [TestMethod]
+        public void CanNavigateFromUsersToHome()
+        {
+            Tester.UITest((Driver) =>
+            {
+                var rootUri = new Uri(RootUrl);
+                Driver.Navigate().GoToUrl(new Uri(rootUri, UsersPage.Slug));
+
+                UsersPage usersPage = new UsersPage(Driver);
+                IWebElement homeLink = usersPage.HomeLink;
+
+                homeLink.Click();
+
+                Assert.AreEqual(RootUrl, Driver.Url);
+            });
+        }
+
+        private UsersPage CreateUser(IWebDriver Driver, string firstName, string lastName)
+        {
+            var rootUri = new Uri(RootUrl);
+            Driver.Navigate().GoToUrl(new Uri(rootUri, UsersPage.Slug));
+            var page = new UsersPage(Driver);
+            page.AddUser.Click();
+
+            var addUserPage = new AddUsersPage(Driver);
+
+            addUserPage.UserFirstNameTextBox.SendKeys(firstName);
+            addUserPage.UserLastNameTextBox.SendKeys(lastName);
+            addUserPage.SubmitButton.Click();
+            return page;
+        }
+    }
+
+    public class UITester
+    {
+        private IEnumerable<IWebDriver> Drivers;
+        private string FolderName { get; set; }
+        public UITester(IEnumerable<IWebDriver> drivers, string folderName = "Screenshots")
+        {
+            Drivers = drivers;
+            FolderName = folderName;
+        }
+
+        public void UITest(Action<IWebDriver> action)
+        {
+            foreach (IWebDriver webDriver in Drivers)
+            {
+                try
+                {
+                    action(webDriver);
+                }
+                catch (Exception ex)
+                {
+                    string fileName = $"{action.Method.Name}-{webDriver.GetType()}.png";
+                    fileName = string.Join("", fileName.Split(Path.GetInvalidFileNameChars()));
+                    string currentDirectory = Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName;
+                    string path = Path.Combine(currentDirectory, FolderName, fileName);
+
+                    Screenshot screenshot = ((ITakesScreenshot)webDriver).GetScreenshot();
```

Comment:
This works because the current driver supports screenshots, but what happens when the Driver doesn't implement ITakesScreenshot? Better to type check rather than simply cast.
```suggestion
                if (Driver is ITakesScreenshot takesScreenshot)
                {
                    var screenshot = takesScrrenshot.GetScreenshot();
```